### DISCUSSION TITLE
HEAD is now at c1a9ce62a Merge pull request #13163 from Homebrew/dependabot/bundler/Library/Homebrew/rubocop-rspec-2.10.0

### DISCUSSION THREAD

**Discussion by Cleevemadjouh** [Homebrew/discussions#3202] [Everyday usage]
Please someone could help me to install Homebrew ?

MacBook-Pro ~ % /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
==> Checking for `sudo` access (which may request your password)...
Password:
Sorry, try again.
Password:
==> This script will install:
/opt/homebrew/bin/brew
/opt/homebrew/share/doc/homebrew
/opt/homebrew/share/man/man1/brew.1
/opt/homebrew/share/zsh/site-functions/_brew
/opt/homebrew/etc/bash_completion.d/brew
/opt/homebrew

Press RETURN/ENTER to continue or any other key to abort:
==> /usr/bin/sudo /usr/sbin/chown -R cleevemadjouh:admin /opt/homebrew
==> Downloading and installing Homebrew...
HEAD is now at c1a9ce62a Merge pull request #13163 from Homebrew/dependabot/bundler/Library/Homebrew/rubocop-rspec-2.10.0

**Comment by SMillerDev** [comment]
What is the issue? That looks like a normal install.

**Reply by Cleevemadjouh** [reply]
MacBook-Pro ~ % brew --version
Homebrew 3.4.7-23-gc1a9ce6
Homebrew/homebrew-core (no Git repository)
MacBook-Pro ~ % brew install git
Running `brew update --preinstall`...
error: Not a valid ref: refs/remotes/origin/master
fatal: ambiguous argument 'refs/remotes/origin/master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

**Reply by SMillerDev** [reply]
What does `brew doctor` say?

**Reply by Cleevemadjouh** [reply]
MacBook-Pro ~ % brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Homebrew/homebrew-core was not tapped properly! Run:
  rm -rf "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core"
  brew tap homebrew/core

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Library/Frameworks/Python.framework/Versions/3.9/bin/python3.9-config
  /Library/Frameworks/Python.framework/Versions/3.9/bin/python3-config

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/core) checkout 

Warning: Broken symlinks were found. Remove them with `brew cleanup`:
  /usr/local/bin/code
  /usr/local/bin/easy_install-3.9

**Reply by SMillerDev** [reply]
Could you try this part:
```
Warning: Homebrew/homebrew-core was not tapped properly! Run:
rm -rf "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core"
brew tap homebrew/core
```

**Reply by Cleevemadjouh** [reply]
Last login: Sat Apr 23 07:44:09 on ttys000
MacBook-Pro ~ % rm -rf "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core"
MacBook-Pro ~ % brew doctor
Error: No available tap homebrew/core.
MacBook-Pro ~ % brew doctor
Error: No available tap homebrew/core.
MacBook-Pro ~ % brew --version
Homebrew 3.4.7-23-gc1a9ce6
Homebrew/homebrew-core N/A
MacBook-Pro ~ % brew tap homebrew/core
Running `brew update --preinstall`...
==> Homebrew has enabled anonymous aggregate formula and cask analytics.
Read the analytics documentation (and how to opt-out) here:
  https://docs.brew.sh/Analytics
No analytics have been recorded yet (nor will be during this `brew` run).

==> Homebrew is run entirely by unpaid volunteers. Please consider donating:
  https://github.com/Homebrew/brew#donations
==> Tapping homebrew/core
Cloning into '/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core'...
remote: Enumerating objects: 1180720, done.
remote: Counting objects: 100% (14/14), done.
remote: Compressing objects: 100% (8/8), done.
error: 2899 bytes of body are still expected72 MiB | 174.00 KiB/s
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
Error: Failure while executing; `git clone https://github.com/Homebrew/homebrew-core /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core --origin=origin --template=` exited with 128.
Error: Failure while executing; `/usr/local/bin/brew tap homebrew/core` exited with 1.
==> Tapping homebrew/core
Cloning into '/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core'...
remote: Enumerating objects: 1180720, done.
remote: Counting objects: 100% (14/14), done.
remote: Compressing objects: 100% (8/8), done.
error: RPC failed; curl 56 LibreSSL SSL_read: error:02FFF036:system library:func(4095):Connection reset by peer, errno 54
error: 19 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
Error: Failure while executing; `git clone https://github.com/Homebrew/homebrew-core /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core --origin=origin --template=` exited with 128.
MacBook-Pro ~ %

**Reply by SMillerDev** [reply]
Your connection does not seem good enough to download the homebrew repositories. See #666 for more info

**Reply by Cleevemadjouh** [reply]
I tried again:

MacBook-Pro ~ % brew tap homebrew/core
Running `brew update --preinstall`...
==> Tapping homebrew/core
Cloning into '/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core'...
remote: Enumerating objects: 1180720, done.
remote: Counting objects: 100% (14/14), done.
remote: Compressing objects: 100% (8/8), done.
error: 7578 bytes of body are still expected1.18 MiB | 41.00 KiB/s 
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
Error: Failure while executing; `git clone https://github.com/Homebrew/homebrew-core /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core --origin=origin --template=` exited with 128.
Error: Failure while executing; `/usr/local/bin/brew tap homebrew/core` exited with 1.
==> Tapping homebrew/core
Cloning into '/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core'...
remote: Enumerating objects: 1180760, done.
remote: Counting objects: 100% (47/47), done.
remote: Compressing objects: 100% (29/29), done.
remote: Total 1180760 (delta 26), reused 36 (delta 18), pack-reused 1180713
Receiving objects: 100% (1180760/1180760), 465.45 MiB | 95.00 KiB/s, done.
Resolving deltas: 100% (817831/817831), done.
Tapped 3 commands and 6034 formulae (6,373 files, 511.4MB).
MacBook-Pro ~ % brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Library/Frameworks/Python.framework/Versions/3.9/bin/python3.9-config
  /Library/Frameworks/Python.framework/Versions/3.9/bin/python3-config
MacBook-Pro ~ %

**Reply by apramm** [reply]
Hey! I had similar problem. What I did was first uninstalling the No Github Rep. homebrew version already in my mac using : /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/uninstall.sh)" 
then, installed again. That worked for me but i don't know why it wasn't installed properly in first go

**Reply by valluriavi** [reply]
<img width="1033" alt="Screenshot 2022-09-28 at 1 04 23 PM" src="https://user-images.githubusercontent.com/67725161/192717077-ce3ffa81-5d76-4ea9-9683-958cb7515a20.png">

Wait for few minutes and its installing faced similar issue.

**Reply by AlDenty** [reply]
> Hey! I had similar problem. What I did was first uninstalling the No Github Rep. homebrew version already in my mac using : /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/uninstall.sh)" then, installed again. That worked for me but i don't know why it wasn't installed properly in first go

This is what worked for me when I had this issue. Thank you!

**Reply by andersong9012** [reply]
This one worked for me. Thanks

**Reply by deabllaca** [reply]
Worked for me as well !

**Comment by omartheman** [comment]
For me, installation hung at that point for a minute or two. After waiting, installation completed. Full installation input/output was this (Note your message from above at about the 15th line. Also note the two commands at the end that must be run before `brew` can be used.): 

```
omarshishani@MacBook-Pro-3 ~ % /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
==> Checking for `sudo` access (which may request your password)...
Password:
==> This script will install:
/opt/homebrew/bin/brew
/opt/homebrew/share/doc/homebrew
/opt/homebrew/share/man/man1/brew.1
/opt/homebrew/share/zsh/site-functions/_brew
/opt/homebrew/etc/bash_completion.d/brew
/opt/homebrew

Press RETURN/ENTER to continue or any other key to abort:
==> /usr/bin/sudo /usr/sbin/chown -R omarshishani:admin /opt/homebrew
==> Downloading and installing Homebrew...
HEAD is now at cc6119261 Merge pull request #13449 from Homebrew/update-man-completions
Updated 1 tap (homebrew/core).
Warning: /opt/homebrew/bin is not in your PATH.
  Instructions on how to configure your shell for Homebrew
  can be found in the 'Next steps' section below.
==> Installation successful!

==> Homebrew has enabled anonymous aggregate formulae and cask analytics.
Read the analytics documentation (and how to opt-out) here:
  https://docs.brew.sh/Analytics
No analytics data has been sent yet (nor will any be during this install run).

==> Homebrew is run entirely by unpaid volunteers. Please consider donating:
  https://github.com/Homebrew/brew#donations

==> Next steps:
- Run these two commands in your terminal to add Homebrew to your PATH:
    echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' >> /Users/omarshishani/.zprofile
    eval "$(/opt/homebrew/bin/brew shellenv)"
- Run brew help to get started
- Further documentation:
    https://docs.brew.sh

omarshishani@MacBook-Pro-3 ~ % 
```

**Comment by csa7mdm** [comment]
https://stackoverflow.com/questions/39836190/homebrew-install-failed-during-git-fetch-origin-masterrefs-remotes-origin-mas

(remove the "cellar" and "homebrew" directories from /usr/local)
